crypto/getenv.c

```diff
@@ -15,6 +15,50 @@
 #include "internal/cryptlib.h"
 #include "internal/e_os.h"
 
+/*
+ * Additional checks for securely allowing access to selected environment variables
+ */
+#if defined (OPENSSL_NETCAP_ALLOW_ENV) && defined(__linux__)
+
+/*
+ * Utility function to check if a string is in an array of strings. returns true if found
+ */
+int str_in_list(const char *candidate, const char *list[], int n)
+{
+  if ( (candidate!=NULL) && (list!=NULL))
```

Comment:
Rather than having "arrow code", why not test inputs and do an early return

```
if (candidate == NULL || list == NULL || len < 1)
    return 0;
```